### ISSUE TITLE
Maintain cluster-wide logical replication slots in SynchronousReplication mode

### ISSUE THREAD

**Issue by ivlcic** [sorintlab/stolon#819]
**What would you like to be added**:
Maintaining an online logical replication slots state for **SynchronousReplication** so the logical replication could switch-over when standby is promoted. 

**Why is this needed**:
This is needed for logical replication external to Stolon cluster. 
For instance: Data warehousing solution with Stolon cluster Hot data -> Logical rep. -> PostgreSQL Cold data)

**Naive proposal**
Some double checking must me done since I'm no sure this is possible. 
Maybe one solution would be creating logical repl. slot on all servers with periodically calling pg_replication_slot_advance  
on standbys reading master's replication slot state (taken in consideration that SynchronousReplication must be enabled - no lag on standbys).

It might be that its impossible to make it correctly:
[Logical replication and physical standby failover](https://wiki.postgresql.org/wiki/Logical_replication_and_physical_standby_failover)

**Comment by sebasmannem** [comment]
Yes, your proposed solution is also what Patroni does. And what EFM does as well. 
If we want it in Stolon, maybe I can issue a PR for it.

**Comment by ivlcic** [comment]
That would be great. My knowledge of PostgreSQL and Stolon internals is insufficient to do it.

**Comment by Ondrysak** [comment]
Some relevant information on how Patroni is doing this

https://www.postgresql.eu/events/pgconfde2022/sessions/session/3745/slides/306/Implementing%20failover%20of%20logical%20replication%20slots%20in%20Patroni.pdf
